packages/components/src/components/TableComponent/TableComponent.stories.js

```diff
@@ -0,0 +1,156 @@
+        date: text('Date', '100 years ago')
+      }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={dataRows}
+        dataHeaders={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        handleNew={action('handleNew')}
+        handleDelete={action('handleDelete')}
+      />
+    );
+  });
+
+  /*
+  HOW TO USE:
+  - Required Props:
+      - title: String name of the Resource being surfaced
+
+      - dataHeaders: Array containing each column (in form of object)
+        appearing in data table.
+        Example:
+          [
+            { key: 'name', header: 'Name' },
+            { key: 'namespace', header: 'Namespace' },
+            { key: 'date', header: 'Date Modified' }
+          ]
+
+      - dataRows: Array containing each row (in form of object)
+        appearing in data table.
+        *all keys in dataHeaders must be specified in each row object*
+          Example:
+            [
+              {
+                **id: 'namespace1:resource-one',
+                name: 'resource-one',
+                namespace: 'namespace1',
+                date: '16 minutes ago'
+              }
+            ]
+          ** Must be specified for React to know which rows have changed,
+            are added, or are removed. It is also used when trying to
+            figure out which row(s) are about to be deleted. Finally, must
+            follow above pattern: {NAMESPACE}:{NAME_OF_RESOURCE} **
+
+      - loading: Boolean Retrieved from Redux store and passed as prop.
+
+      - selectedNamespace: String Retrieved from Redux store and passed as prop.
+        Used when no rows are available and allows us to tell the user in which
+        namespace.
+
+    Optional Props:
+      - HandleNew: Handler used when Add Button is clicked and if not specifiied,
+        Add Button will not appear.
+
+      - HandleDelete: Handler used when Delete Button is clicked and if not
+        specifiied, Delete Button and checkboxes will not appear. Must take 2 arguments,
+        selected rows and cancel batch action to unselect rows once deletion is
+        finished. It is recommended to save batchAction as state for future use.
+          Example:
+
+          rowsInput will be passed as:
+            [
+              {
+                name: resource-one
+                namespace: namespace1
+              }
+            ]
+
+          HandleDelete = (rowsInput, cancelBatchAction) => {
+            this.setState({
+              * other state changes *
+              userSelectedRows: rowsInput,
+              ** unselectRows: cancelBatchAction
+            });
+          }
+
+        ** calling this.state.unselectRows() when desired resources are deleted
```

Comment:
Are you saying when unselectRows is provided here, it will go ahead and unselect the rows? Shouldn't this always happen really (why's it optional? Only if delete is present perhaps?)?

Reply:
based on Alan's comment ill change this approach. But basically the thing is that when you click delete button the table stays with the rows checked and highlighted even after you've done the "deletion". So to bypass this one must call getBatchActionProps().onCancel() to reset the table to normal view. But this getBatchActionProps is defined within the table so to use it either call it within the table or pass it as an argument so it can be used later one if saved as a state.